mmlearn/tasks/classification.py

```diff
@@ -0,0 +1,250 @@
+            for spec in self.task_specs:
+                assert Modalities.has_modality(spec.query_modality)
+                query_modality = Modalities.get_modality(spec.query_modality)
+                mode = self.mode
+                metric_name = spec.metric_name
+                metric = getattr(torchmetrics, metric_name)
+                num_classes = all_dataset_info[dataset_name].get_class_count()
+                task = "multiclass" if num_classes > 2 else "binary"
+                self.metrics.update(
+                    {
+                        (
+                            query_modality,
+                            mode,
+                            metric_name,
+                            dataset_name,
+                        ): MetricCollection(
+                            {
+                                f"{query_modality}_{mode}_C_{metric_name}@{k}_{dataset_name}": (
+                                    metric(
+                                        **(
+                                            {
+                                                "top_k": k,
+                                                "task": task,
+                                                "num_classes": num_classes,
+                                            }
+                                            if "top_k"
+                                            in inspect.signature(metric).parameters
+                                            else {
+                                                "task": "multiclass",
+                                                "num_classes": num_classes,
+                                            }
+                                        )
+                                    ).to("cuda")
+                                )
+                                for k in spec.top_k
+                            }
+                        )
+                    }
+                )
+
+    def evaluation_step(
+        self,
+        trainer: Trainer,
+        pl_module: LightningModule,
+        batch: Dict[str, torch.Tensor],
+        batch_idx: int,
+    ) -> None:
+        """Update classification accuracy metrics."""
+        if trainer.sanity_checking:
+            return
+
+        outputs: Dict[Union[str, Modality], Any] = pl_module(batch)
+
+        for (
+            query_modality,
+            _,
+            _,
+            dataset_name,
+        ), metric in self.metrics.items():
+            output_embeddings = outputs[query_modality.embedding]  # Predictions
+            label_index = batch[query_modality.target]  # True labels
+            names = batch[
+                NAME_KEY
+            ]  # List of dataset names for each element in the batch
+
+            # Filter indices where dataset name is part of the metric_name
+            matching_indices = [
+                i for i, name in enumerate(names) if name in dataset_name
+            ]
```

Comment:
There is `example_index`, dataset_index` and `example_ids` (which is a tuple of example_index and dataset_index) that can help you identify samples from the same dataset. Is the names key really necessary?